app/models/user.rb

```diff
@@ -0,0 +1,49 @@
+# == Schema Information
+#
+# Table name: users
+#
+#  id                     :integer          not null, primary key
+#  email                  :string           default(""), not null
+#  encrypted_password     :string           default(""), not null
+#  reset_password_token   :string
+#  reset_password_sent_at :datetime
+#  remember_created_at    :datetime
+#  sign_in_count          :integer          default(0), not null
+#  current_sign_in_at     :datetime
+#  last_sign_in_at        :datetime
+#  current_sign_in_ip     :string
+#  last_sign_in_ip        :string
+#  created_at             :datetime         not null
+#  updated_at             :datetime         not null
+#  provider               :string
+#  uid                    :string
+#  role                   :string           default("coach"), not null
+#
+
+class User < ActiveRecord::Base
+  # Include default devise modules. Others available are:
+  # :confirmable, :lockable, :timeoutable and :omniauthable
+  devise :database_authenticatable, :recoverable, 
+    :rememberable, :trackable, :validatable, :omniauthable, :omniauth_providers => [:github]
+
+  ADMIN = "admin"
+  COACH = "coach"
+
+  def self.from_omniauth(auth)
```

Comment:
this method is a bit long - I'll still pull in but it would be good to refactor to make it easier to see what the components are doing